qt_ui/windows/QLiberationWindow.py

```diff
@@ -474,6 +504,7 @@ def closeEvent(self, event: QCloseEvent) -> None:
             self._save_window_geometry()
             super().closeEvent(event)
             self.dialog = None
+            self.debriefing = None
```

Comment:
Force debriefing to go out of scope, otherwise the loop over `QApplication.topLevelWidgets()` will close the debriefing window again which may trigger the end-game message for a second time. Mind that it shouldn't be possible to close the main window when a debriefing window is still open because the debriefing window forces the focus on itself for as long as it is open. This means we can let the debriefing go out of scope safely.

Reply:
You can't assume anything about when this will be garbage collected.

Reply:
> You can't assume anything about when this will be garbage collected.

That's not the point... This should assure that the end-game popup doesn't show itself a second time, because it will if you leave this line out.

Either way, once I reviewed everything and all changes are applied, I'll verify if this is still an issue.

Reply:
Conclusion is the following: removing the `closeEvent` method in `QDebriefingWindow` fixes the "double pop-up" issue, however as mentioned in the related conversation this does introduce its own set of problems.

Reply:
```suggestion
            if hasattr(self, 'debriefing'):
                del self.debriefing
```

If this makes you happier, it's an alternative I've tested which also works...